### ISSUE TITLE
[Bug] Broken Tests on main after #235

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/pocket#308]
## Objective

Fix broken bugs on main introduced by #235.

## Origin Document

#235 was a large PR that reintroduced a bug fixed by #282.

Specifically, in `runtime/test_artifacts/uti.go`, the addition of `os.Exit(0)`

```go
func CleanupPostgresDocker(_ *testing.M, pool *dockertest.Pool, resource *dockertest.Resource) {
	// You can't defer this because `os.Exit`` doesn't care for defer
	if err := pool.Purge(resource); err != nil {
		log.Fatalf("could not purge resource: %s", err)
	}
	os.Exit(0)
}
```

Ignores the `exitCode` in `persistence/test/setup_test.go`

```go
func TestMain(m *testing.M) {
	pool, resource, dbUrl := test_artifacts.SetupPostgresDocker()
	testPersistenceMod = newTestPersistenceModule(dbUrl)
	exitCode := m.Run()
	test_artifacts.CleanupPostgresDocker(m, pool, resource)
	os.Exit(exitCode)
}
```

Resulting in various errors committed to main:

![Screen Shot 2022-10-18 at 10 12 52 AM](https://user-images.githubusercontent.com/1892194/196499447-de3341bc-4d4c-4cc4-b668-6c6e90b285e0.png)

Doing a hard reset to the prior commit (`git reset --hard 2c390c3c055b4d1a36fb93075867bcf96b89fdcd`) validated that these tests were functional in the previous commit.

## Goals

- `make develop_test` should pass after removing `os.Exit(0)` from `runtime/test_artifacts/util.go`

## Deliverable

- [ ] Removal of `os.Exit(0)` in `runtime/test_artifacts/util.go`
- [ ] Fixes in all broken tests raised by `make test_all`

## Non-goals / Non-deliverables

- New functionality
- Refactoring / organizing existing code

## Testing Methodology

- **All tests**: `make develop_test`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @Olshansk 
**Co-Owners**: @deblasis
